src/rubrix/labeling/text_classification/weak_labels.py

```diff
@@ -0,0 +1,395 @@
+
+        for n, record in tqdm(
+            enumerate(self._records), total=len(self._records), desc="Applying rules"
+        ):
+            # First: fill annotation array
+            annotation = record.annotation or "None"
+
+            try:
+                annotation = _label2int[annotation]
+            except KeyError as error:
+                # When a label2int was provided, we want to raise an error if the label is missing!
+                if label2int is not None:
+                    raise MissingLabelError(
+                        f"The annotation label '{annotation}' is missing in the `label2int` dict {label2int}"
+                    ) from error
+                # we already have "None" -> we need to subtract 1
+                _label2int[annotation] = len(_label2int) - 1
+                annotation = _label2int[annotation]
+
+            annotation_array[n] = annotation
+
+            # Second: fill weak label matrix
+            for m, rule in enumerate(self._rules):
+                weak_label = rule(record) or "None"
+
+                try:
+                    weak_label = _label2int[weak_label]
+                except KeyError as error:
+                    if label2int is not None:
+                        raise MissingLabelError(
+                            f"A rule returned the weak label '{weak_label}', "
+                            f"but it is missing in the `label2int` dict {label2int}"
+                        ) from error
+                    _label2int[weak_label] = len(_label2int) - 1
+                    weak_label = _label2int[weak_label]
+
+                weak_label_matrix[n, m] = weak_label
+
+        return weak_label_matrix, annotation_array, _label2int
+
+    @property
+    def rules(self) -> List[Callable]:
+        """The rules (labeling functions) that were used to produce the weak labels."""
+        return self._rules
+
+    @property
+    def records(self) -> List[TextClassificationRecord]:
+        """The records corresponding to the weak labels."""
+        return self._records
+
+    @property
+    def label2int(self) -> Dict[str, int]:
+        """The dictionary that maps the weak/annotation labels to integers."""
+        return self._label2int
+
+    @property
+    def matrix(self) -> np.ndarray:
+        """The weak label matrix."""
+        return self._matrix
+
+    def train_matrix(self) -> np.ndarray:
+        """Returns the part of the weak label ``self.matrix`` that has NO corresponding annotations.
+
+        Returns:
+            The train weak label matrix.
+        """
+        return self._matrix[self._annotation_array == self._label2int["None"]]
+
+    def test_matrix(self) -> np.ndarray:
+        """Returns the part of the weak label ``self.matrix`` that has corresponding annotations.
+
+        Returns:
+            The test weak label matrix.
+        """
+        return self._matrix[self._annotation_array != self._label2int["None"]]
```

Comment:
Discussing about annotated data  in rubrix with @dvsrepo, this filter could not be a validated test set. I mean
we could have annotated data in rubrix but still non validated, so, this matrix could be confusing for the whole process.

What do you think @dvsrepo @dcfidalgo ?

Reply:
For now, we will leave these shortcuts for the most common use case, that is annotated records are actually validated. The shortcuts are explicitly explained in the docstrings. But the user can always create their own masks for the label matrix/annotations:
```python
maks = [rec.status == "Validated" for rec in weak_labels.records]
weak_labels.matrix[mask]
```